app/routes/synonyms/synonyms.server.ts

```diff
@@ -148,7 +181,19 @@ export async function putSynonyms({
   additions?: string[]
   subtractions?: string[]
 }) {
+  const response = {
+    success: false,
+    synonymId,
+    error: null,
+  } as PutSynonymResponse
   if (!subtractions?.length && !additions?.length) return
+  if (additions?.length) {
+    const isValid = await validateEventIds({ eventIds: additions })
+    if (!isValid) {
+      response.error = 'Invalid eventId'
+      return response
+    }
+  }
```

Comment:
Raise an exception if the input is invalid. If you want to be RESTfully correct, then raise a 400 Bad Request error.

Reply:
You do not want the banner that Judy requested? I believe the request was for a banner to alert the moderator that an eventId was incorrect so they could check the input and re-enter it. I could just make it 400 but in that case it would not have the banner telling the user why it failed.
<img width="1108" alt="Screenshot 2024-08-16 at 11 41 56 AM" src="https://github.com/user-attachments/assets/abd5df3d-e04c-4884-b7a3-be347abdca37">

Reply:
What does it mean for an eventId to be invalid?

Reply:
that it does not appear as the eventId on any circular in the database. In the above image, all of the eventIds exist except "nope". Since "nope" was not found to be an eventId on any circular, that eventId is invalid. I can remove the validation so that a moderator could put anything, but I was worried that in that particular scenario they might add something they thought was correct but was mistyped. So what I was trying to accomplish there was a "hey, you entered an eventId that doesn't exist anywhere, that means what you think you linked to the group would not include any circulars. check over these ids below and see if you entered something wrong and then try saving again."

Reply:
If you just want it to throw an error response, I can try to make that informative in a way that gives some indication of what went wrong, but I haven't explored those options yet, so I am not sure yet what that would look like.

Reply:
I can also adapt the message so that it is much clearer what an invalid eventId is

Reply:
> that it does not appear as the eventId on any circular in the database. In the above image, all of the eventIds exist except "nope". Since "nope" was not found to be an eventId on any circular, that eventId is invalid. I can remove the validation so that a moderator could put anything, but I was worried that in that particular scenario they might add something they thought was correct but was mistyped. So what I was trying to accomplish there was a "hey, you entered an eventId that doesn't exist anywhere, that means what you think you linked to the group would not include any circulars. check over these ids below and see if you entered something wrong and then try saving again."

How about we have a combo box that auto-completes from existing event IDs, and the submit button is disabled until you choose a valid one?

@dakota002 may have refactored our combo box code to the point that it is reasonably reusable.

Reply:
I can definitely look into that.